opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -378,6 +378,12 @@ def test_default_span_resource(self):
         # pylint: disable=protected-access
         self.assertIs(span.resource, resources._EMPTY_RESOURCE)
 
+    def test_span_context_remote_flag(self):
```

Comment:
maybe add a docstring saying that the expected behavior is is_remote is False by default? I guess it's implicit, I just like having some human description so I can verify the intended behavior.

Reply:
Actually, I think there should not be a default for `is_remote`. And if any, the default should be `true`. This flag might have security implications, e.g. samplers may trust a sampled=True flag on a is_remote=False SpanContext. If the is_remote flag is wrongly False, this might lead to a Denial of Service vulnerability.

Reply:
I can definitely see that argument, although I'm wondering if denial of service is a concern, there isn't a more robust way of dealing with that. This also assumes that the adapters (include internal/custom ones) also properly use the is_remote flag, which is impossible to enforce.

Reply:
on the other hand, I feel like is_remote=True as the default is unusual behavior, as the remote span is much more of a special case, and currently users are creating/starting spans with no arguments, which would then make then remote spans by default unless one is aware of that flag, and sets it everywhere.

We may need more opinions here, but I think the default being false ensures minimal friction for usage of the opentelemetry APIs.

Reply:
Sorry, this was a misunderstanding from me! 🙈 
I thought we were talking about manually creating a SpanContext. For starting a new span, having is_remote=False on its SpanContext is the only sane thing to do (and this should *not* be configurable).